### PR TITLE
CI Fixes

### PR DESCRIPTION
**Pull request by mikemhenry**

### CODE REVIEW

**Comment by IAlibay** [comment]
Was about to write on the other PR - I don't think 2018's rdkit releases supported > py 3.7, so I'd probably just move that one out of the matrix and put it as a single runner (i.e. py3.7 + 2018.03.4)?

**Comment by IAlibay** [comment]
Also I'd skip out on py3.10, rdkit doesn't support it yet (annoyingly it's blocking MDA's full move to py3.10).

**Comment by mikemhenry** [comment]
No more issues with missing files but things like this:

https://github.com/OpenFreeEnergy/Lomap/runs/5015302658?check_suite_focus=true#step:8:653
`AttributeError: 'QueryAtom' object has no attribute 'SetQuery'`

Is odd since it should have that attribute 
https://www.rdkit.org/docs/source/rdkit.Chem.rdchem.html?highlight=queryatom#rdkit.Chem.rdchem.QueryAtom

**Comment by IAlibay** [comment]
Ok I had a bit of a deeper dive here the problem seems to be that SetQuery didn't get added until late 2018: https://github.com/rdkit/rdkit/pull/2088 so I assume after the 2018 release was completed.

However that code path wasn't added until 2019 on lomap (https://github.com/OpenFreeEnergy/Lomap/commit/878efeefcd32b8b8e257a3a802fd245fac6f95bb)

So we're kinda doing a chicken and the egg type of thing.

There's two options here -

1) merge as-is and then start moving towards rdkit 2021 (things will break but c'est la vie)
2) we revert some of the cresset changes to lomap, get to point where it passes CI, and then bump up the rdkit version.

How have you been approaching this until now @richardjgowers ?

**Comment by mikemhenry** [comment]
I'd vote to merge as-is, then create a new PR that adds rdkit 2021 to the testing matrix to catch things broken by api changes.

I do think we should mark these tests as okay to fail, or skip them, so that we can be sure we don't break things in future PRs, since right now its a pain to go through and see what is failing. 

What should we do about the other test failures? I think they were mostly issues with score differences, which I am not sure where those are coming from.

**Comment by IAlibay** [comment]
Yeah I'd be happy with merge as-is too. I'll let @richardjgowers make the call though.

> What should we do about the other test failures? I think they were mostly issues with score differences, which I am not sure where those are coming from.

I suspect those are also caused by erm... recent contributions to lomap that might not have respected the rdkit pins. Maybe the best thing to do is start picking at each test individually until everything passes?

File: .github/workflows/CI.yaml [hunk 1]
```diff
@@ -29,8 +29,8 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        python-version: [3.8, 3.9, "3.10"]
-        rdkit-version: ["2021.09", "2021.03", "2018.03.4"]
+        python-version: [3.7]
+        rdkit-version: ["2018.03.4"]
```
Comment (richardjgowers) [inline .github/workflows/CI.yaml:33]: I'd rather we kept the latest rdkit as our gold standard (even if this means things break for now).  I think I'd prefer this matrix to be `[2018, 2021]` to cover what used to work and what should work respectively.  Then whatever the py needs to be to get those two builds working

**Review by richardjgowers — CHANGES_REQUESTED** [review]
split rdkit in 2 to get both retro + future views?

**Comment by mikemhenry** [comment]
@richardjgowers well this is wild but it is all passing now :upside_down_face:

**Review by richardjgowers — APPROVED** [review]

**Pull request merged**